File: Lab3/t03_06_e3966.py
def bin_search(arr, n, el):
    l = 0
    r = n - 1
    while l <= r:
        m = l + (r - l) // 2
        if el > arr[m]:
            l = m + 1
        elif el < arr[m]:
            r = m - 1
        else: return "YES"
    return "NO"


n = int(input())
col = list(map(int, input().split(" ")))
m = int(input())
inp = input().split(" ")

for i in range(m):
    try:
        mes = bin_search(col, n, int(inp[i]))
        print(mes)
    except:
        print("NO")

File: Lab3/t03_06_e3966.cpp
#include <iostream>
using namespace std;

int bin_search(int arr[], int n, int el){
    int l = 0;
    int r = n - 1;
    while (l <= r){
        int m = l + (r - l) / 2;
        if (el > arr[m]){
            l = m + 1;
        }
        else if (el < arr[m]){
            r = m - 1;
        }
        else{
            cout << "YES\n";
            return 0;
        }
    }
    cout << "NO\n";
    return 0;      
}

int main(){
    int n, m;
    cin >> n;
    int col[n];
    for(int i = 0; i < n; i++){
        cin >> col[i];
    }
    cin >> m;
    int inp[m];
    for(int i = 0; i < m; i++){
        cin >> inp[i];
    }

    for(int i = 0; i < m; i++){
        bin_search(col, n, inp[i]);
    }
    return 0;
}
